### PR TITLE
Out of memory error template. Approach #2

### PR DESCRIPTION
**Pull request by Piankero**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by Piankero** [comment]
@markus2330 is this approach more fitting for you? (the build server will most likely will fail but I would correct all messages as soon as it is clear which way we go)

File: src/plugins/lua/lua.cpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ int elektraLuaOpen (ckdb::Plugin * handle, ckdb::Key * errorKey)
 	/* init new lua state */
 	if ((data->L = lua_newstate (Lua_alloc, NULL)) == NULL)
 	{
-		ELEKTRA_SET_OUT_OF_MEMORY_ERROR (errorKey, "Unable to create new lua state");
```
Comment (markus2330) [inline src/plugins/lua/lua.cpp]: better make this resource error

**Comment by markus2330** [comment]
Yes, looks better now, only the size of the allocation might be of relevance as parameter.

**Comment by Piankero** [comment]
> Open for me is if there is a "size" in all cases.

It looks like the majority has a "rough estimation" available which could be integrated. But concerning the jvm I do not see a possibility unfortunately: 

https://github.com/ElektraInitiative/libelektra/blob/cbbb2436f7e13322661d09525e9c4a51de44ebb6/src/plugins/jni/jni.c#L229-L231

**Comment by markus2330** [comment]
> But concerning the jvm I do not see a possibility unfortunately:

Yes, you are right. Then lets forget about the sizes. Please do not forget to remove ELEKTRA_MALLOC_ERROR

**Comment by Piankero** [comment]
jenkins build libelektra please

**Comment by Piankero** [comment]
jenkins build libelektra please

File: src/plugins/conditionals/conditionals.c [hunk 2]
```diff
@@ -617,17 +616,15 @@ static CondResult parseConditionString (const Key * meta, const Key * suffixList
...
 		// possible error would be out of
 		// memory
 		ksDel (ks);
 		return ERROR;
 	}
 	if ((ret = regcomp (&regex2, regexString2, REGEX_FLAGS_CONDITION)))
 	{
-		ELEKTRA_SET_OUT_OF_MEMORY_ERROR (parentKey,
-						 "Couldn't compile regex: most likely out of memory"); // the regex compiles so the only
+		ELEKTRA_SET_OUT_OF_MEMORY_ERROR (parentKey); // the regex compiles so the only
```
Comment (markus2330) [inline src/plugins/conditionals/conditionals.c]: This is wrongly classified (at least at one place). `regerror()` tells what the error is. (see also above and below). Also remove the comment.

**Comment by markus2330** [comment]
Please fix the one wrong classification in the next PR.

**Pull request merged**